### DISCUSSION TITLE
Consistent capitalisation on public/private keys

### DISCUSSION THREAD

**Discussion by Fastidious** [FiloSottile/age#240] [General]
## What were you trying to do

I generated a new public/private key.

## What happened

Keys are generated, but the public key have all lower case letters, whilst the private key have all capitalised. Can they both be set to be capitalised—or lowercased, it doesn't matter, I am looking for consistency.

**Comment by alerque** [comment]
Kind of useful to be able to spot when somebody has sent you the wrong key just from the copy/paste but without any trappings labeling it.

**Comment by Fastidious** [comment]
@alerque private keys start with `AGE-SECRET-KEY-`, and the public starts with `age`. 😳

I would have used: `ASK-key-all-caps` for the secret key, and `APK-key-all-caps` for the public key.

**Comment by str4d** [comment]
I prefer that public keys are lowercased and secret keys are uppercased. It provides an additional useful visual indicator to users that they are different and should be treated differently, and we cannot have too many of them IMHO (particularly given age's focus on making life easy for users).

I also like the explicit long prefix for secret keys; they do not need to be copied around nearly as much, and being explicit here is better for usability.

All that being said, Bech32 can be uppercased or lowercased freely, and Bech32 decoders will accept both, so anyone can store keys however they want.
